contracts/UniversalChannelHandler.sol

```diff
@@ -0,0 +1,171 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IbcDispatcher.sol";
+import "./IbcMiddleware.sol";
+import {IbcReceiver} from "./IbcReceiver.sol";
+import "./Ibc.sol";
+
+contract UniversalChannelHandler is IbcReceiverBase, IbcUniversalChannelMW {
+    constructor(IbcDispatcher _dispatcher) IbcReceiverBase(_dispatcher) {}
+
+    bytes32[] public connectedChannels;
+    string constant VERSION = "1.0";
+    uint256 public constant MW_ID = 1;
+
+    // Key: middleware bitmap, Value: middleware address from receiver(chain B)'s perspective
+    mapping(uint256 => address[]) public mwStackAddrs;
```

Comment:
Was thinking we'd just pass this in from Polymer?

Also, we'd have to register every combination like this.